### PR TITLE
Fix error message when removing only child from GraphNode

### LINKED ISSUE #89695: Error message when removing only child from GraphNode
### Tested versions

Reproducible in 4.3.dev5, 4.2.1 stable
Not Reproducible in 4.1.1 and 4.0.2

### System information

Windows 10

### Issue description

Whenever I try removing a child from a GraphNode when the removing child is the only child, an error message is issued in the output:
Screenshot 2024-03-19 175626

I would expect this error not to appear when removing the only child.

This issue had been encountered and fixed before when it caused the editor to crash: [#69284](https://github.com/godotengine/godot/pull/69284)

Since then, it appears to have returned, but it fortunately does not crash the editor.

### Steps to reproduce

1. Create a new GraphNode.
2. Add a child node to the GraphNode.
3. Delete the child node created in step 2.
4. Observe the error in the Output.

### Minimal reproduction project (MRP)

[min_repro_project.zip](https://github.com/godotengine/godot/files/14657868/min_repro_project.zip)

### PR DESCRIPTION
**Pull request by ghost**
Pull Request for issue regarding error thrown on delete of only child of a graph node.

*Bugsquad edit:*
- Fixes #89695

### CODE REVIEW

File: scene/gui/graph_node.cpp [hunk 1]
```diff
@@ -336,7 +336,8 @@ void GraphNode::_notification(int p_what) {
 
 			int width = get_size().width - sb_panel->get_minimum_size().x;
 
-			if (get_child_count() > 0) {
+			// Take into account the HboxContainer child
+			if (get_child_count() > 1) {
```
Comment (akien-mga) [inline scene/gui/graph_node.cpp:340]: Would be better to use the boolean argument to exclude internal nodes: https://github.com/godotengine/godot/blob/79de2eaae7fb3bb8f13031ac2ec9b19128b6f471/scene/main/node.h#L431
Then the comment is likely not needed as it's bound to become outdated in the future if internal children change.

**Comment by Geometror** [comment]
Thanks for looking into this!
There's just one thing: Please remove the issue reference in your commit message to prevent spamming the issue with reference updates :)

File: tests/scene/test_graph_node.h [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+TEST_CASE("[SceneTree][GraphNode]") {
+	SUBCASE("[GraphNode][Only Child] Graph Node only child on delete should not cause error.") {
+        //Setup
+		GraphNode *test_node = memnew(GraphNode);
+        test_child->set_name("Graph Node");
+		Control *test_child = memnew(Control);
+		test_child->set_name("child");
+		test_node->add_child(test_child);
+
+        //Test
```
Comment (akien-mga) [inline tests/scene/test_graph_node.h:50]:
```suggestion
        // Setup.
		GraphNode *test_node = memnew(GraphNode);
        test_child->set_name("Graph Node");
		Control *test_child = memnew(Control);
		test_child->set_name("child");
		test_node->add_child(test_child);

        // Test.
```

File: tests/scene/test_graph_node.h [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        CHECK_NOTHROW_MESSAGE(test_node->remove_child(test_child));
```
Comment (AThousandShips) [inline tests/scene/test_graph_node.h:51]:
```suggestion
		// Setup.
		GraphNode *test_node = memnew(GraphNode);
		test_child->set_name("Graph Node");
		Control *test_child = memnew(Control);
		test_child->set_name("child");
		test_node->add_child(test_child);

		// Test.
		CHECK_NOTHROW_MESSAGE(test_node->remove_child(test_child));
```

Comment (akien-mga) [inline tests/scene/test_graph_node.h:51]:
```suggestion
		CHECK_NOTHROW_MESSAGE(test_node->remove_child(test_child));
```

File: tests/scene/test_graph_node.h [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+		memdelete(test_child);
```
Comment (akien-mga) [inline tests/scene/test_graph_node.h:53]: You should only need to delete `test_node`, which should delete its child nodes too.

**Review by akien-mga — APPROVED** [review]
@teacher: I met your student today and confirm this was a top-notch first contribution to an open source project of the scale of Godot ;)

(Also do reach out, I'm interested in your program to get students contributing to open source ;))

**Comment by akien-mga** [comment]
Amended the commit to address review feedback.

**Comment by akien-mga** [comment]
Thanks! And congrats for your first merged Godot contribution :tada:

**Pull request merged**